rp2040-hal/src/clocks/macros.rs

```diff
@@ -107,17 +166,31 @@ macro_rules! clock {
                         clock_nr: pac::clocks::clk_ref_ctrl::SRC_A::CLKSRC_CLK_REF_AUX as u8,
                     }
                 }
+            }
+
+            impl Clock for $name {
+                type Variant = [<$reg:camel SrcType>];
+
+                #[doc = "Get operating frequency for `"$name"`"]
+                fn freq(&self) -> Hertz {
+                    self.frequency
+                }
 
-                /// Configure this clock based on a clock source and desired frequency
-                pub fn configure_clock<S:ClockSource + ValidSrc<$name, Variant=[<$reg:camel SrcType>]>>(&mut self, src: &S, freq: Hertz) -> bool{
-                    let src_freq: Hertz  = src.get_freq();
+                #[doc = "Configure `"$name"`"]
+                fn configure_clock<S: ValidSrc<$name>>(&mut self, src: &S, freq: Hertz) -> bool{
+                    let src_freq: Hertz<u64> = src.get_freq().into();
 
-                    if freq .gt(& src_freq){
+                    if freq.gt(&src_freq){
                         return false;
```

Comment:
See above, this should be something like `Err(ClockError::CantIncrease)` or whatever. Don't have a strong opinion on the name.

Reply:
Done